examples/seq2seq/finetune_trainer.py

```diff
@@ -0,0 +1,463 @@
+    set_seed(training_args.seed)
+
+    # Load pretrained model and tokenizer
+    #
+    # Distributed training:
+    # The .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        cache_dir=model_args.cache_dir,
+    )
+    tokenizer = AutoTokenizer.from_pretrained(
+        model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
+        cache_dir=model_args.cache_dir,
+    )
+    model = AutoModelForSeq2SeqLM.from_pretrained(
+        model_args.model_name_or_path,
+        from_tf=".ckpt" in model_args.model_name_or_path,
+        config=config,
+        cache_dir=model_args.cache_dir,
+    )
+
+    # use task specific params
+    use_task_specific_params(model, data_args.task)
+
+    # set num_beams for evaluation
+    if data_args.eval_beams is not None:
+        model.config.num_beams = data_args.eval_beams
+    assert model.config.num_beams >= 1, f"got eval_beams={model.config.num_beams}. Need an integer >= 1"
+
+    # set max length for generation
+    model.config.max_generate_length = data_args.val_max_target_length
+
+    # set decoder_start_token_id for MBart, TODO(@sshleifer): use_task_specific_params for this
+    if model.config.decoder_start_token_id is None and isinstance(tokenizer, MBartTokenizer):
+        decoder_start_token_id = tokenizer.lang_code_to_id[data_args.tgt_lang]
+        model.config.decoder_start_token_id = decoder_start_token_id
+
+    def build_compute_metrics_fn(task_name: str) -> Callable[[EvalPrediction], Dict]:
+        def non_pad_len(tokens: np.ndarray) -> int:
+            return np.count_nonzero(tokens != tokenizer.pad_token_type_id)
+
+        def decode_pred(pred: EvalPrediction) -> Tuple[List[str], List[str]]:
+            pred_str = tokenizer.batch_decode(pred.predictions, skip_special_tokens=True)
+            label_str = tokenizer.batch_decode(pred.label_ids, skip_special_tokens=True)
+            pred_str = lmap(str.strip, pred_str)
+            label_str = lmap(str.strip, label_str)
+            return pred_str, label_str
+
+        def summarization_metrics(pred: EvalPrediction) -> Dict:
+            pred_str, label_str = decode_pred(pred)
+            rouge: Dict = calculate_rouge(pred_str, label_str)
+            summ_len = np.mean(lmap(len, pred.predictions))
+            rouge.update({"gen_len": summ_len})
+            return rouge
+
+        def translation_metrics(pred: EvalPrediction) -> Dict:
+            pred_str, label_str = decode_pred(pred)
+            bleu: Dict = calculate_bleu(pred_str, label_str)
+            gen_len = np.mean(lmap(non_pad_len, pred.predictions))
+            bleu.update({"gen_len": gen_len})
+            return bleu
+
+        compute_metrics_fn = summarization_metrics if task_name == "summarization" else translation_metrics
```

Comment:
should this be rather 
```python
compute_metrics_fn = summarization_metrics if  "summarization" in task_name else translation_metrics
```

because we pass `summarization_{dataset}` for pegasus